### ISSUE TITLE
*ocTransferId* directories in 'files_versions' with content from 'Shared'

### ISSUE THREAD

**Issue by abika** [owncloud/core#17386]
The setup is a on Owncloud server for a small company. They have two folders shared with read/write access for all employees and sync these with desktop clients for Windows and Mac.

Here it comes: Recently weird folders pop up under `files_versions` containing the `Shared` folder again:
`owncloud/data/*user*/files_versions/Shared/shared_folder/subfolder/*somefile.txt*ocTransferId766080870.part/Shared/shared_folder/...`
And now the content of `shared_folder` comes again, and again and again...
`.../subfolder/*somefile.txt*ocTransferId766080870.part/Shared/shared_folder/subfolder/*somefile.txt*ocTransferId766080870.part/Shared/shared_folder/subfolder/*somefile.txt*ocTransferId766080870.part/Shared/shared_folder/...`

(*user* is the one whos the initial owner of `shared_folder` and is sharing it with everybody else)

This happened multiple times with multiple files and as the shared files are copied many, many times a lot of server disc space gets wasted (already deleted 40GB manually, whereas the total amount of files is only 15GB).

To be honest, I have now idea what the cause for the creation of these folder are but it must be some heavy bug.

**Comment by LukasReschke** [comment]
Thanks a lot for reporting issues back to us!

We need some more information to properly support you with your issue!

Can I ask you to follow our guidelines for submitting bugs as described here: https://github.com/owncloud/core/blob/master/CONTRIBUTING.md

**Comment by abika** [comment]
Sorry, didn't see that. Unfortunately I can't reproduce anything since I'm only administrating the server, not using it.

### Server configuration
**Operating system**: Ubuntu 12.04.5

**Web server:** Apache/2.2.22 (Ubuntu)

**Database:** sqlite 3.7.9

**PHP version:** 5.4.42-1+deb.sury.org~precise+1

**ownCloud version:** 8.0.4 (but happened already with 7.0.4)

**List of activated apps:** Files, Activity, Pictures

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => 'ocd87ec0f849',
  'passwordsalt' => '...',
  'trusted_domains' =>
  array (
    0 => '....net',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'sqlite3',
  'version' => '8.0.4.2',
  'installed' => true,
  'maxZipInputSize' => 10485760000,
  'allowZipDownload' => true,
  'forcessl' => true,
  'theme' => '',
  'maintenance' => false,
  'secret' => '...',
  'mail_smtpmode' => 'php',
  'loglevel' => 0,
);
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Logs
#### Web server error log

-nothing suspicious here-

#### ownCloud log (data/owncloud.log)

Well it's 9MB big, I found one entry for the directory that makes problems:
```
{"reqId":"89f219396861296d952a0acc6b3ffb9f","remoteAddr":"84.189.5.161","app":"PHP","message":"copy(\/var\/www\/owncloud\/data\/julia\/files_versions\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/PurchaseOrder Templates\/Bestellung Grafik rechts.dot.ocTransferId1398867476.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/PurchaseOrder Templates\/Bestellung Grafik rechts.dot.ocTransferId1398867476.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/PurchaseOrder Templates\/Bestellung Grafik rechts.dot.ocTransferId1398867476.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Finanzen\/Lexware\/Templates\/OrderConfirmation Templates\/Auftragsbest\u00e4tigung Grafik rechts.ott.ocTransferId766080870.part\/Shared\/EOI_GF\/Angebote\/BA\/15401_D04_Eigenerkl\u00e4rung Zuverl\u00e4ssigkeit_Gesetzestreue.doc.ocTransferId2032156293.part\/Shared\/EOI_GF\/Angebote\/Drosos\/15 06 01 TVI DE 24 Ausschreibung Evaluation Kapazit\u00e4tsaufbau.pdf.ocTransferId1246803410.part\/_GF\/GmbH Akten\/20140721_Vorschlag_sGmbH.docx.v1406060074): failed to open stream: Invalid argument at \/var\/www\/owncloud\/lib\/private\/files\/storage\/local.php#211","level":3,"time":"2015-07-01T20:08:52+00:00","method":"MOVE","url":"\/owncloud\/remote.php\/webdav\/Shared\/StatEval_GF\/Finanzen\/Lexware\/Templates"}
```
If you need more from log, tell me what I should look for

**Comment by PVince81** [comment]
Wow, this is super weird.

It looks like the part files are used as folders ?!

Do you happen to have symbolic links or hard links inside the data folder ?

**Comment by abika** [comment]
> It looks like the part files are used as folders ?!

Exactly, and the recursion is even more weird. Can somebody explain what the actual purpose of these "filename".ocTransferIdxxx.part files is? My guess is these are temporary files during syncing, but should they stay after a sync? There are also a lot of them under 'files', e.g. for the "problem file" in question:
```
-rw-r--r--  1 www-data www-data  49K Mär 11 11:47 Bestellung Grafik rechts.dot
-rw-r--r--  1 www-data www-data    0 Mär 11 11:41 Bestellung Grafik rechts.dot.ocTransferId1398867476.part
-rw-r--r--  1 www-data www-data    0 Mär 11 11:32 Bestellung Grafik rechts.dot.ocTransferId1895893780.part
```
> Do you happen to have symbolic links or hard links inside the data folder ? 

`$/var/www/owncloud# find . -type l` : nothing
`$/var/www/owncloud# find -type f -links +1`  : nothing

[ Guess I will upgrade to 8.1 soon. ]

**Comment by PVince81** [comment]
The part files are temporary files during an upload.

Basically it works this way (simple example):
1. Client PUTs a file "test.txt" with WebDAV
2. Server stores the file in the target directory, for example "data/user1/files/target/path/test.txt.ocTransferId12345.part"
3. Once the transfer is finished, "test.txt.ocTransferId12345.part" is renamed to "test.txt" (and might overwrite if a file exists with this name)

If the files did not get deleted or renamed, it means that the final rename didn't work for some reason.
Since 8.1 the part files should properly get removed in case of errors.

I heard people saying that part files might be moved to the trashbin in some cases when deleted (fixed in 8.1).

But in older versions, even if the part file stays there or goes to trash, I don't understand how they could land in "files_versions" ! And even weirder is that they become folders !

**Comment by PVince81** [comment]
Is this file "Bestellung Grafik rechts.dot" the only one with this issue ? Or are they more ?

You could also try and find the matching entries in the database:
`select * from oc_filecache where path like '%Bestellung Grafik rechts%' order by path`
Maybe it will contain more clues.

**Comment by abika** [comment]
Right now this is the only one, as I kept it consuming 3GB space. But I removed similar directories (like 6 or so) last week each as of them were 5GB big.

```
sqlite> select * from oc_filecache where path like '%Bestellung Grafik rechts%' order by path;
fileid|storage|path_hash|parent|name|mimetype|mimepart|mtime|storage_mtime|encrypted|etag|path|permissions|size|unencrypted_size
38894|10|79cf6427d1c19ef96fbf68a8f55d9367|38890|Bestellung Grafik rechts.dot|16|3|1426070858|1426070858|0|a65dcb16a9f499f718dbba1a3fb56858|files/Shared/EOI_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.dot|27|49664|0
38892|10|225fa672787a126f44037b4ad15bbad0|38890|Bestellung Grafik rechts.ott|14|3|1426070522|1426070522|0|dce98127c37049305fcbdbdfb28fb360|files/Shared/EOI_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.ott|27|45741|0
39576|5|44f064df0dbdf71be6acd648b89cae85|39544|Bestellung Grafik rechts.dot|16|3|1426070858|1426070858|0|2bf2655587a3aef024124c34da040bc5|files/StatEval_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.dot|27|49664|0
39578|5|6abd703d7bfad5bbc9ab798389589694|39544|Bestellung Grafik rechts.ott|14|3|1426070522|1426070522|0|704e9e16c4f2d892bc339ff3f6246d57|files/StatEval_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.ott|27|45741|0
39198|10|582e85cdcc46c968be67f5ba7e908200|39194|Bestellung Grafik rechts.dot|16|3|1426070858|1426070858|0|4e3662f4e0eea230d0c40d19bdb1dad7|files_trashbin/files/Templates.d1435780767/PurchaseOrder Templates/Bestellung Grafik rechts.dot|27|49664|0
39340|5|582e85cdcc46c968be67f5ba7e908200|39323|Bestellung Grafik rechts.dot|16|3|1426070858|1426070858|0|a9faaa2a6750364def5a9f08aa945131|files_trashbin/files/Templates.d1435780767/PurchaseOrder Templates/Bestellung Grafik rechts.dot|27|49664|0
39196|10|90090c7d860f4015de5ee4bcc330ef1e|39194|Bestellung Grafik rechts.ott|14|3|1426070522|1426070522|0|44635988c4be90b5aa4ef7816191012a|files_trashbin/files/Templates.d1435780767/PurchaseOrder Templates/Bestellung Grafik rechts.ott|27|45741|0
39342|5|90090c7d860f4015de5ee4bcc330ef1e|39323|Bestellung Grafik rechts.ott|14|3|1426070522|1426070522|0|2ecd27e08afb74744f82a2062ce0ba5e|files_trashbin/files/Templates.d1435780767/PurchaseOrder Templates/Bestellung Grafik rechts.ott|27|45741|0
38929|10|651dbeca1ec7a2753418e3ab11d688e2|38907|Bestellung Grafik rechts.dot.v1426070858|14|3|1435778782|1435778782|0|dcc87c19405fdb60f2d8ab3dd878be56|files_versions/Shared/EOI_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.dot.v1426070858|27|49664|0
38911|10|a35a9a2861ef15d846a4e422996b3487|38907|Bestellung Grafik rechts.ott.v1426070522|14|3|1435777103|1435777103|0|61d881844a2458a2cabdb6a9d418b809|files_versions/Shared/EOI_GF/Finanzen/Lexware/Templates/PurchaseOrder Templates/Bestellung Grafik rechts.ott.v1426070522|27|45741|0
```

**Comment by PVince81** [comment]
Okay, so there is no trace of the "part" entries in the database.

Maybe let's check the share entry, is that a shared file ?
`select * from oc_share where file_target in (38894, 38892, 39576, 39578)`

I find it strange that people have the "Shared" folder directly. It's probably the old one. I hope you never manually cleared "oc_filecache" before ?

**Comment by abika** [comment]
`sqlite> select * from oc_share where file_target in (38894, 38892, 39576, 39578);
sqlite>`
nothing
> I find it strange that people have the "Shared" folder directly

Yes, one user has files there (he's using a mac). Empty for everybody else
```$/var/www/owncloud/data# ls -lh */files/Shared
alex/files/Shared:
total 8,0K
drwxr-xr-x 5 www-data www-data 4,0K Jun 24 12:26 EOI_GF
drwxr-xr-x 3 www-data www-data 4,0K Jun 20 17:32 EOI_Team
...```
> I hope you never manually cleared "oc_filecache" before ?

Nope, dont even know where it is.

**Comment by PVince81** [comment]
Some weird sharing issues have been fixed lately in 8.1.3, you might want to try again with that version

**Comment by PVince81** [comment]
Not sure what kind of glitch caused this situation. Please try again with 8.2.2 or 9.0 RC1 where many bugfixes versions went in.

If you still see such entries appear, feel free to reopen.
